lib/matplotlib/_fontconfig_pattern.py

```diff
@@ -125,14 +125,10 @@ def parse(self, pattern):
         props = self._properties = {}
         try:
             self._parser.parseString(pattern)
-        except self.ParseException as e:
-            raise ValueError(
-                "Could not parse font string: '%s'\n%s" % (pattern, e)) from e
-
+        except self.ParseException as err:
+            raise ValueError("\n" + err.explain(0)) from None
```

Comment:
I can't judge how much trouble bumping the pyparsing dependency might cause. If we want to be defensive, we could do 

```suggestion
            # can be simplified to err.explain(0) when pyparsing>=3
            message = ParseException.explain(err, 0)
            raise ValueError("\n" + message) from None
```

which works for both versions pyparsing 2 and 3.

Reply:
Sure, went back to requiring 2.3.1.